### DISCUSSION TITLE
[Feature] Partner sharing with options for more privileges

### DISCUSSION THREAD

**Discussion by silmaril42** [immich-app/immich#13206] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This should go on the list in #12614.

Currently, when using partner sharing, the partner sees all images, but there is no search functionality and they don't see the  people that have been found in the photos.

I think this was a wise choice to minimize the risk of leaking information to other users, but it also limits the use-cases of this feature.

So I'd like to see some options for each partner share (all defaulting to False). 
The ones I can think of now are
- Share people information
- Allow searching
- Allow editing people information (naming or hiding people)
- Allow editing file meta-data
- Allow deleting files
- Allow adding files

This would enable us to make good choices about the level of teamwork we choose, while still having a clear separation of users.

My current workaround is to use a shared account with my wife to manage our collection of >20k photos, but I'd prefer a clear separation of privileges per person. Then we could share the same collection with our kids, who would get all read privileges so they can search for people, dates or whatever they like without any way to damage existing data.

Some of this might also be applied to album sharing.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Nairou** [comment]
Came here looking for this. I spent an hour trying to figure out why I couldn't see people tagged in certain photos, and got errors trying to add them, when it turned out they were shared photos (not obvious when partner timeline sharing is enabled).

**Comment by Green-Kite** [comment]
I'm hoping for more privileges for partners or people in a shared album too. I want to let others edit pictures in the album and even delete the pictures out of a shared album.

**Comment by maxmar391** [comment]
In the context of partner sharing, I would also like to see that tags can be shared. This would be a great addition, for example when your partner and you are working on selecting from all photos for a project, like a physical photo album or defining a selection of  pictures to print out. So please enable tags sharing.

**Comment by Se7enair** [comment]
I would also add the functionality to decide which assets should be shared trough the partner sharing.

At the moment I use three accounts to be able to have a "own" account for my wife and me and a account with all shared assets. therefore I need a possibilty to move assets between accounts to decide a picture should be shared trough partner sharing or not. Or the ability to decide which assets should be shared with my partner.

A shared album is also not a good idea, because it also does not allowing me to edit metadata like persons etc.

**Reply by Se7enair** [reply]
Also sharing tags would be nice.

**Comment by orzechszek** [comment]
Good idea - more partner sharing options should be available

**Comment by phlepper** [comment]
I'll add my $0.02:

I have a library of photos I want to share with extended family.  I would like them (as partners) to have access to all the photos as if they were theirs: edit faces, edit metadata, etc.  I just don't want them to be able to add or delete photos.  Right now, the only way I've found to do this is to have everyone use the admin account, which isn't great.

Thanks!
Paul

**Comment by TKaluza** [comment]
Absolut important feature. Very high WAF for switching from Google Photos.

**Comment by bo0tzz** [comment]
Dupe of #12614
